staging/src/k8s.io/mount-utils/mount_helper_unix.go

```diff
@@ -52,7 +52,7 @@ func IsCorruptedMnt(err error) bool {
 		underlyingError = pe.Err
 	}
 
-	return underlyingError == syscall.ENOTCONN || underlyingError == syscall.ESTALE || underlyingError == syscall.EIO || underlyingError == syscall.EACCES
+	return underlyingError == syscall.ENOTCONN || underlyingError == syscall.ESTALE || underlyingError == syscall.EIO || underlyingError == syscall.EACCES || underlyingError == syscall.EHOSTDOWN
```

Comment:
I wonder though - if we should separate corrupted mounts from socket/host not connected errors. But may be we could consider them same. cc @chakri-nelluri

Reply:
When the host is down does that actually mean the mount is gone and cleaned up on the client side? Wondering if this will cause stale/leaking mounts and we need to do something like a force unmount instead.

Reply:
Here is the way how this `host is down` mount point is unmounted in CSI driver, SMB CSI driver invoke `CleanupMountPoint` func to unmount mount point:
 - without this PR, `PathExists` return `false, err` directly, and then `CleanupMountPoint` returns `Warning: Unmount skipped because path does not exist`
 - with this PR, `PathExists` return `true, err`, and finally invoke `doCleanupMountPoint` to unmount, that will fix this issue.

So by this PR, CSI driver should unmount corrupted mount point if using `CleanupMountPoint` func.  @jingxu97 

https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/mount-utils/mount_helper_common.go#L137-L147
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/mount-utils/mount_helper_common.go#L31-L41

Reply:
how about Windows case? Is windows version of function can already handle this case?

Reply:
@jingxu97 good point, I added windows fix.
While this specific SMB unmount issue actually does not exists on Windows since `CleanupMountPoint` on Windows just removes directly. Anyway, should also fix `IsCorruptedMnt` issue on Windows.
```
func CleanupMountPoint(m *mount.SafeFormatAndMount, target string, extensiveMountCheck bool) error {
	proxy, ok := m.Interface.(*mounter.CSIProxyMounter)
	if !ok {
		return fmt.Errorf("could not cast to csi proxy class")
	}
	return proxy.Rmdir(target)
}
```